components/api/db/cellery_hub_db.bal

```diff
@@ -340,32 +389,95 @@ returns json | error {
     return check json.convert(imagesListResponse);
 }
 
-public function updateImage(string orgName, string imageName, string description, string userId) returns sql:UpdateResult | error? {
-    log:printInfo(io:sprintf("Performing update on image : %s in organization %s", imageName, orgName));
-    sql:UpdateResult res = check connection->update(UPDATE_IMAGE_QUERY, description, imageName, orgName, userId);
+public function updateImageDescriptionNSummary(string orgName, string imageName, string description, string summary, string userId) returns sql:UpdateResult | error? {
+    log:printInfo(io:sprintf("Updating description and summary of the image %s/%s", orgName, imageName));
+    sql:UpdateResult res = check connection->update(UPDATE_IMAGE_DESCRIPTION_N_SUMMARY_QUERY, description, summary, imageName, orgName, userId);
     return res;
 }
 
+public function updateImageKeywords(string orgName, string imageName, string[] keywords, string userId) returns error? {
+    log:printInfo(io:sprintf("Updating keywords of the image %s/%s", orgName, imageName));
+    string role = check getUserRole(userId, orgName);
+    if role == constants:ROLE_ADMIN || role == constants:ROLE_PUSH {
+        log:printInfo(io:sprintf("User %s is allowd to update the image %s/%s", userId, orgName, imageName));
+        string imageId = check getArtifactImageID(orgName, imageName);
+        if imageId != "" {
+            _ = check connection->update(DELETE_IMAGE_KEYWORDS_QUERY, imageId);
+            log:printInfo(io:sprintf("Successfully deleted keywords of the image %s/%s", orgName, imageName));
+
+            string[][] dataBatch = [];
+            int i = 0;
+            foreach var keyword in keywords {
+                if keyword != "" {
+                    dataBatch[i] = [imageId, keyword];
+                    i = i + 1;
+                }
+            }
+            if i > 0 {
+                _ = check connection->batchUpdate(INSERT_IMAGE_KEYWORDS_QUERY, ...dataBatch);
+                log:printInfo(io:sprintf("Successfully inserted keywords for the image %s/%s", orgName, imageName));
+            } else {
+                log:printDebug(io:sprintf("No keywords found. Hence not performing keyword insertion for image %s/%s", orgName, imageName));
+            }
+        } else {
+            string errMsg = io:sprintf("Unable to update image %s/%s. Artifact Image Id of the image is not found", orgName, imageName);
+            log:printDebug(errMsg);
+            error er = error(errMsg);
+            return er;
+        }
+    } else {
+        string errMsg = io:sprintf("User %s is not allowd to update keywords of the image %s/%s", userId, orgName, imageName);
+        log:printDebug(errMsg);
```

Comment:
This should be an error log.

Reply:
Remove this log because of skipping the role validation